### PR TITLE
Remove lib auth

### LINKED ISSUE #666: Annotate magic properties on Account.
E.g.
@property-read int $confirmed

see https://scrutinizer-ci.com/g/BitLucid/ninjawars/inspections/b54110a3-2bf5-45c6-a523-20cc776b8503/issues/files/deploy/lib/control/SignupController.php?status=new&selectedSeverities%5B0%5D=10&orderField=path&order=asc&honorSelectedPaths=0#inspectioncomment-16637497

Two other member vars here:
https://scrutinizer-ci.com/g/BitLucid/ninjawars/inspections/b54110a3-2bf5-45c6-a523-20cc776b8503/issues/files/deploy/lib/data/Account.php?status=new&selectedSeverities%5B0%5D=10&orderField=path&order=asc&honorSelectedPaths=0

Apparently also some ?new? ones introduced on the Player object.

https://scrutinizer-ci.com/g/BitLucid/ninjawars/inspections/b54110a3-2bf5-45c6-a523-20cc776b8503/issues/files/deploy/lib/control/SignupController.php?status=new&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&fileId=deploy%2Flib%2Fcontrol%2FSignupController.php&honorSelectedPaths=0

### PR DESCRIPTION
**Pull request by Beagle**
Yeah, that's right. lib_auth is *gone*!

Fixes #666

### CODE REVIEW

File: deploy/lib/control/AccountController.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function showChangeEmailForm() {
      */
     public function changeEmail() {
         // confirm_delete
-        $player     = Player::find(self_char_id());
+        $player     = Player::find(SessionFactory::getSession()->get('player_id'));
```
Comment (tchalvak) [inline deploy/lib/control/AccountController.php]: We need something better than that.

Comment (Beagle) [inline deploy/lib/control/AccountController.php]: We will at some point.

**Comment by tchalvak** [comment]
Maybe Controllers should get a base inherited controller and we should set the logged_in auth on the controller, and then reference it as a member variable or method `$this->login_id()` in all controller contexts?  And then we'd only pull it from the session once, in the router.  Obviously non-controller contexts would be a different set of exceptions.

Also, I'm thinking a `SessionFactory::login_id()` would be a better 

Also creating an issue for examining whether we should move to something other than `player_id` in the session, `account_id` for example.
https://github.com/BitLucid/ninjawars/issues/685

File: deploy/lib/control/NewsController.php [hunk 2]
```diff
@@ -19,42 +19,47 @@ class NewsController {
...
 
     /**
      * Check whether a player has the necessary create role
+     *
      * @return boolean
      */
-    private function hasCreateRole($pc){
-        if(!($pc instanceof Player)){
+    private function hasCreateRole($pc) {
+        if (!($pc instanceof Player)) {
```
Comment (tchalvak) [inline deploy/lib/control/NewsController.php]: Not the CODING_STANDARD , see docs/CODING_STANDARD

**Comment by tchalvak** [comment]
Since this deals with both signup & login, have you tested whether those still work?

**Comment by Beagle** [comment]
Tested login, not sign up

File: deploy/lib/control/AccountController.php [hunk 3]
```diff
@@ -95,7 +95,7 @@ public function showChangePasswordForm() {
      * Change account password
      */
     public function changePassword() {
-        $player     = Player::find(self_char_id());
+        $player     = Player::find(SessionFactory::getSession()->get('player_id'));
```
Comment (tchalvak) [inline deploy/lib/control/AccountController.php]: Creating an issue for alternatives to `SessionFactory::getSession()->get('player_id');`

**Comment by tchalvak** [comment]
Login seems to be broken on this branch.

screenshot from 2016-04-07 11 44 26

**Comment by Beagle** [comment]
Woot!                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        From: Roy R.Sent: Friday, April 8, 2016 19:11To: BitLucid/ninjawarsReply To: BitLucid/ninjawarsCc: BeagleSubject: Re: [BitLucid/ninjawars] Remove lib auth (#678)Merged #678.

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub

**Pull request merged**